### PR TITLE
Background Polling for Test Result

### PR DESCRIPTION
**Pull request by AndroidMedaGalaxy**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [ ] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [ ] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [ ] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) (if applicable)
* [ ] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [X] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)

## Description
Implementation of  Background polling to check for test Result and raise local notification

Please refer to https://jira.itc.sap.com/browse/EXPOSUREAPP-893

### CODE REVIEW

**Comment by jakobmoellersap** [comment]
Please ensure that the CI Pipe runs :)

File: Corona-Warn-App/config/detekt.yml [hunk 1]
```diff
@@ -100,7 +100,7 @@ complexity:
     ignoreStringsRegex: '$^'
   TooManyFunctions:
     active: true
-    excludes: ['**/test/**', '**/androidTest/**', '**/*.Test.kt', '**/*.Spec.kt', '**/*.Spek.kt', '**/LocalData.kt', '**/formatter/*Helper.kt']
+    excludes: ['**/test/**', '**/androidTest/**', '**/*.Test.kt', '**/*.Spec.kt', '**/*.Spek.kt', '**/LocalData.kt', '**/formatter/*Helper.kt', '**/worker/BackgroundWorkScheduler.kt']
```
Comment (jakobmoellersap) [inline Corona-Warn-App/config/detekt.yml:103]: Why do we need this here? Can we split up the scheduler code?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    workerParams: WorkerParameters
+) :
+    CoroutineWorker(context, workerParams) {
+
+    companion object {
+        private val TAG: String? = DiagnosisTestResultRetrievalPeriodicWorker::class.simpleName
+    }
+
+    /**
+     * Work execution
```
Comment (jakobmoellersap) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/DiagnosisTestResultRetrievalPeriodicWorker.kt:34]: This method should really have more comments.

**Review by jakobmoellerdev — CHANGES_REQUESTED** [review]
Some Remarks, other than that it looks good from my side.

File: Corona-Warn-App/config/detekt.yml (hunk 1 above)
Comment (ritsam) [reply Corona-Warn-App/config/detekt.yml:103]: @AlexanderAlferov What are your thoughts on this?

Comment (ChristopherSchmitz) [reply Corona-Warn-App/config/detekt.yml:103]: @AlexanderAlferov  could you please comment here. We have to close this PR tonight (CC: @jakobmoellersap )

**Review by jakobmoellerdev — APPROVED** [review]

**Review by AlexanderAlferov — APPROVED** [review]

**Pull request merged**